fragment.go

```diff
@@ -0,0 +1,122 @@
+package vite
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+)
+
+// HTMLFragment generates an HTML fragment for Vite integration based on the provided configuration.
+//
+// This function takes a Config struct and uses it to create the necessary HTML
+// elements for including Vite-managed assets in a web page. The returned HTML
+// is typically intended to be placed in the <head> section of an HTML document.
+//
+// Parameters:
+//   - config: A Config struct containing the necessary configuration options for Vite integration.
+//
+// Returns:
+//   - template.HTML: An HTML fragment as a template.HTML type, which can be safely embedded in HTML templates.
+//   - error: An error if the HTML fragment generation fails for any reason.
+//
+// The returned HTML fragment may include elements such as:
+//   - Script tags for Vite client in development mode
+//   - Links to stylesheets
+//   - Module preload tags
+//   - Script tags for entry points and other necessary JavaScript
+//
+// Usage example:
+//
+//	fragment, err := HTMLFragment(myConfig)
+//	if err != nil {
+//	    // Handle error
+//	}
+//	// Use fragment in your HTML template
+func HTMLFragment(config Config) (template.HTML, error) {
```

Comment:
I wonder if we should return a struct instead of the `template.HTML` directly.

```go
type Fragment struct {
  Scripts template.HTML
}
```

For now, the `template.HTML` is good enough. But returning a struct wouldn't introduce a breaking change later.

What do you guys think?

Reply:
Yeah I would agree and suggest return a struct of:
```
type Fragment struct {
  Tags template.HTML
}
```
which means in the template you would write:
```
{{ .Vite.Tags }}
```

Good idea @olivere about breaking changes. I only suggest Tags over Scripts because it includes CSS as well as JS scripts.

Reply:
Yeah, naming is hard. Tags is OK for me.

Reply:
Besides Tags, what props should we include in the struct?

Reply:
For now, I think `Tags` is just fine. What I had in mind is that we might want to provide the users the option to put their scripts in the `<head>` or before the `</body>`. That'd be possible if we split the tags between `script` and `link`. But maybe I'm overengineering stuff, again. :-)

Reply:
I've added this struct in [1b5dfe8](https://github.com/olivere/vite/pull/10/commits/1b5dfe8692d00e8a869f8b31ad6e427c21d05ba5)